### PR TITLE
v3

### PR DESCRIPTION
**Pull request by sowasred2012**
## Done

- Updated global-nav to work specifically with the Vanilla navigation pattern, rather than as a standalone element

## QA

- View the demo
- On desktop, see that the "All Canonical" nav item is present, and that the dropdown appears when clicked
- On mobile, see that "All Canonical" is _not_ present, and instead "Products", "Also From Canonical", "Resources" and "About" dropdowns are visible, and work
- Accessibility: the dropdown should comply with the Vanilla a11y docs: https://vanillaframework.io/docs/patterns/navigation/accessibility

Fixes: https://github.com/canonical-web-and-design/web-squad/issues/4989

### CODE REVIEW

**Comment by sowasred2012** [comment]
@davegoddard42 @lyubomir-popov I've assigned you both to this for a review - I'm not sure what we want to do about the mobile view of this, as it is it's what we had in the global nav previously but contained within the navigation element, but it doesn't look particularly "Vanilla", so I'm looking for some guidance on how it should look: https://global-nav-218.demos.haus/

For more context/background, this is a chat I had with @lyubomir-popov and @bartaz about it in Mattermost: https://chat.canonical.com/canonical/pl/5bcdcwperb8tdnbkryi4xhuf3y

**Comment by davegoddard42** [comment]
Thanks @sowasred2012. Let me look at this more depth, and we can catch up on Monday @lyubomir-popov. We have a little more time now that the launch has been pushed back a month, so let's get it right.

**Comment by lyubomir-popov** [comment]
another alternative would be to have chevrons pointing right and so the nested items as views that slide right to left

**Comment by anthonydillon** [comment]
## QA
- To be fussy - at w260 you lose the Product dropdown and the mobile menu has not kicked in yet.
- Should Esc close the mega menu? Hard to know how to close the menu when blind

File: src/js/global-nav.js [hunk 1]
```diff
@@ -203,61 +196,57 @@ function createProductDropdown(products) {
...
     // we are hiding dropdown content after the animation
     // to prevent it from being focusable
     // 500ms is hardcoded here, which should be enough for
     // most of Vanilla animation speeds
-    setTimeout(() => {
-      forEach(dropdownContents, (_, menu) => {
-        menu.classList.add('u-hide');
-      });
-    }, 500);
+    // setTimeout(() => {
```
Comment (anthonydillon) [inline src/js/global-nav.js:232]: Left in commented code

File: src/js/global-nav.js [hunk 2]
```diff
@@ -268,83 +257,64 @@ function addListeners(breakpoint, wrapper) {
...
-        <li class="global-nav__header-link has-dropdown">
-          <a class="global-nav__header-link-anchor" aria-expanded="false" aria-controls="canonical-products" href="#canonical-products">Products</a>
-        </li>
-      </ul>
-    </div>
-  </div>`);
+  const navItem =
+    createFromHTML(`<li class="p-navigation__item--dropdown-toggle global-nav__dropdown-toggle u-hide--mobile" id="all-canonical">
+      <a href="#canonical-products" aria-controls="canonical-products" class="p-navigation__link global-nav__header-link-anchor ">All Canonical</a>
+    </li><`);
```
Comment (anthonydillon) [inline src/js/global-nav.js:314]:
```suggestion
    </li>`);
```

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Must simpler. Left a few comments in the code.

File: index.html [hunk 3]
```diff
@@ -6,6 +6,7 @@
     <title>Canonical global navigation demo</title>
     <meta name="description" content="Canonical global navigation demo" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="stylesheet" href="https://assets.ubuntu.com/v1/vanilla-framework-version-3.2.0.min.css" />
```
Comment (bartaz) [inline index.html:9]: This seems couple versions behind

Comment (bartaz) [reply index.html:9]: Is it needed at all? Global nav project has Vanilla as a dependency.

Comment (sowasred2012) [reply index.html]: Fixed

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Comment by lyubomir-popov** [comment]
@sowasred2012 all good just one minor cosmetic thing: the space abve the muted heading is way too much. 

image

Can you replicae what happens when you have an hr followed by a muted heading?

image

**Pull request merged**